### PR TITLE
Move duplicate share UIAlertActions to AlertAction struct #359

### PR DESCRIPTION
**Pull request by vanyaland**
#### Summary:
- Create helper `AlertActions` struct for convenient UIAlertAction's instantiation
- Move share actions from `IssuesViewController` and `RepositoryViewController` to `AlertActions `

[Link to origin issue ticket](https://github.com/rnystrom/GitHawk/issues/359)

### CODE REVIEW

**Comment by Sherlouk** [comment]
Looks great! Are you planning on adding other actions as part of this PR? (If so mind whacking WIP in the name until it's ready, easier to manage!)

**Comment by vanyaland** [comment]
Hey, @Sherlouk !

Initially, I saw through issue description and see only 2 cases to move, so I added only this actions, in order to make this changes be atomic and consistent with issue aim.

Of course, I noticed about the other ones. Today I'll add other actions.
If I mismatch any code style guidelines or if you have any better design idea how to wrap the actions, please let me know.

Thanks!

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

File: Classes/Utility/AlertActions.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import UIKit
+
+struct AlertActions {
+    
+    static func share(_ items: [Any],
+                      activities: [UIActivity]?,
+                      controller: UIViewController,
+                      sender: UIBarButtonItem) -> UIAlertAction {
+        weak var weakController = controller
```
Comment (vanyaland) [inline Classes/Utility/AlertActions.swift]: Do I need weak reference here?

**Comment by Sherlouk** [comment]
I'm not seeing 359 reference share directly, in fact the links he has shows all the ones in the same file! Regardless there is more duplication as you identified - appreciate you picking it up!

In terms of a better idea, ultimately it's up to you! In my head I'd like to move all the actions out into one place that builds an action sheet and you just pick which ones to show (and that way a cancel action button is always added if we create new ones in the future)

File: Classes/Utility/AlertActions.swift (hunk 1 above)
Comment (Sherlouk) [reply Classes/Utility/AlertActions.swift]: Probably recommended as we don't want to hold a reference to the Issue/Repo view controllers just for the sake of sharing!

Comment (vanyaland) [reply Classes/Utility/AlertActions.swift]: Ok, thank you!

**Comment by vanyaland** [comment]
Ready for review.

**Comment by rnystrom** [comment]
Hey @vanyaland! Looks like there are some conflicts (sorry! I should have merged sooner). You might want to rebase too, I added a new alert controller to `IssueCommentSectionController` as well.

**Comment by vanyaland** [comment]
Hey, @rnystrom!

- Rebase
- `IssueCommentSectionController`
- `SettingsViewController`

Ready for review.
Thanks!

**Pull request merged**